telegram/ext/_conversationhandler.py

```diff
@@ -708,8 +708,13 @@ def check_update(self, update: object) -> Optional[_CheckUpdateType]:
             # check if future is finished or not
             if state.done():
                 res = state.resolve()
-                self._update_state(res, key)
-                state = self._conversations.get(key)
+                if res is None:
+                    # Special case if the error was raised in a non-blocking entry-point
```

Comment:
so this bug only happens for non blocking entry points and not for handlers in states or fallbacks?

Reply:
Yes. if we're already in a non-None state, then `self._update_state(res, key)` works as expected even if `res` is the old state. We also have a test case for that (right above the newly added one) which covers that. unfortunately it did not cover exceptions in non-blocking entrypoints.